enhancements/machine-config/admin-defined-reboot-policy.md

```diff
@@ -0,0 +1,264 @@
+        value: myservice
+         actions:
+            - type: restart
+              target: myservice
+    clusterDefaultPolicies:
+      - type: file
+        value: /etc/mco/internal-registry-pull-secret.json
+        actions:
+            - type: none
+      - type: file
+        value: /var/lib/kubelet/config.json
+        actions:
+            - type: none
+      - type: file
+        value: /etc/machine-config-daemon/no-reboot/containers-gpg.pub
+        actions:
+            - type: reload
+              target: crio.service
+      - type: file
+        value: /etc/containers/policy.json
+        actions:
+            - type: reload
+              target: crio.service
+      - type: file
+        value: /etc/containers/registries.conf
+        actions:
+            - type: special
+```
+
+Which, when applied, will define actions for all future changes to these MachineConfig fields based on the diff of the update (rendered MachineConfigs).
+
+For this initial implementation we will support MachineConfig changes to:
+ - Files
+ - Units
+ - kernelArguments
+ - sshKeys
+And actions for:
+ - None
+ - Draining a node
+ - Reloading a service
+ - Daemon-reload and restarting of a service
+ - Reboot (default)
+ - Special (see workflow description)
+
+In the future, we can extend this in two ways:
+Supporting more changes/actions (e.g. user defined script to run after a certain change)
+Supporting diffing between ostree images (for image based updates) and bootc managed systems (for bifrost)
+
+
+### Workflow Description
+
+An admin of an OCP cluster can at any time create/update/delete RebootPolicy objects to reconfigure how they wish updates to be applied. There isn’t any immediate effect so long as the RebootPolicy CR itself is a valid object.
+
+The MachineConfigController’s drain controller will process any update to the object and validate the RebootPolicy for duplication, action validity (but not correctness), etc. If the object is valid, the drain controller will update the object’s status and observedGeneration to indicate success, and if not, update those and degrade the controller, example:
+
+```
+status:
+  conditions:
+  - lastTransitionTime: "xxx"
+    Message: “”
+    status: "False"
+    type: RebootPolicyValidationFailing
+  - lastTransitionTime: "xxx"
+    Message: “Validation of the rebootpolicy has succeed.”
+    status: "True"
+    type: RebootPolicyValidated
+  observedGeneration: x
+```
+
+The MachineConfigDaemon will process and apply the current RebootPolicy if it’s valid, indicated by generation == observedGeneration (MCC has processed the latest spec) and RebootPolicyValidated == true/RebootPolicyValidationFailing == false. If not, the daemon will also degrade (and queue a re-sync) waiting for the controller to validate the object.
+
+For now, we only support one cluster-wide rebootpolicy, and as such any rebootpolicy not named “cluster” will be ignored and the MCO will report errors via degrading the operator.
```

Comment:
Typically we can prevent these objects from being created using validating webhooks, or now even a CEL validation. So perhaps restate this to say that CEL validation will prevent creating an object not named `cluster`